pkg/database/user.go

```diff
@@ -330,6 +330,21 @@ func (db *Database) DeleteUser(u *User, actor Auditable) error {
 	})
 }
 
+// PurgeUsers will delete users who are not a system admin, not a member of any realms
+// and have not been modified before the expiry time.
+func (db *Database) PurgeUsers(maxAge time.Duration) (int64, error) {
+	if maxAge > 0 {
+		maxAge = -1 * maxAge
+	}
+	deleteBefore := time.Now().UTC().Add(maxAge)
+	// Delete users who were created/updated before the expiry time.
+	rtn := db.db.Unscoped().
+		Where("users.system_admin = false AND users.created_at < ? AND users.updated_at < ?", deleteBefore, deleteBefore).
```

Comment:
I don't think this does what you think it does? 

- Realm membership changes don't change `updated_at`, so you could end up immediately deleting someone
- That column is updated anytime the direct user record changes, we can happen at login with the revoke check

Reply:
The latter isn't a problem - it's okay to not-delete users who are logging in to keep fresh.

I suppose we can ensure users is bumped on realm-remove

Reply:
> I suppose we can ensure users is bumped on realm-remove

I couldn't figure out a good way to do that. If we can "touch" the update_at on the user when realm membership is changed, I think this is fine (maybe add a comment).

Reply:
I've added `TestRemoveRealmUpdatesTime` and I can't seem to repro your claim that modifying the realm relationship _isn't_ bumping time

Reply:
Interesting... what happens if the user comes serialized from the cache?

Reply:
I believe the reason is that we have wrapped calls with SaveUser here:
https://github.com/google/exposure-notifications-verification-server/blob/f76a3cea09d53d076a47a204ce9c948e183c8f35/pkg/database/user.go#L337

which always saves both the User entry and the associations (even if the table fields haven't changed)